### PR TITLE
[feature] Add debug_dataloader_samples utility to preview decoded dataloader samples (#184)

### LINKED ISSUE #184: Add Debug utility to be able to preview first samples used for training
Ensuring that we're using the correct data is crucial for training and could avoid hours of debugging. It would be cool to have a debug utility that can pull the couple first samples from the dataloader just like we do in trainer, then detokenize them and print them to the user

### PR DESCRIPTION
**Pull request by garongkim**
# What does this PR do?

This PR adds a utility function `debug_dataloader_samples()` under `utils/debug_utils.py`.

It allows users to preview a few decoded examples from the first batch of the training `DataLoader`, which can help ensure that tokenization is working correctly before launching a long training run.  
This debug utility addresses [#184](https://github.com/huggingface/nanotron/issues/184).

Although another contributor expressed interest in the issue, no progress or maintainer feedback has occurred in the last 3 weeks.  
As this feature is simple but useful, I decided to implement and contribute it directly.

### How to use

This utility can be optionally called from `run_train.py` or any other training script to inspect tokenized input samples before training.  
Here is an example snippet:

```python
from nanotron.utils.debug_utils import debug_dataloader_samples

debug_dataloader_samples(train_dataloader, tokenizer, num_samples=3)
```
### Notes

To avoid a circular import issue between `utils.py` and `distributed.py`,  
this function is placed in a new module `debug_utils.py` instead of the main `utils.py`.

Logging is used instead of `print()` to remain consistent with Nanotron’s logging system.

Fixes #184 

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guidelines](https://github.com/huggingface/nanotron/blob/main/CONTRIBUTING.md#submitting-a-new-issue-or-feature-request)?
- [ ] Did you write any new necessary tests?
- [ ] Did you log the throughput and loss you get to ensure the PR works as expected in actual training?
- [ ] Did you log the memory usage? you can use [this tool](https://huggingface.co/spaces/nanotron/predict_memory) to understand the memory usage breakdown in nanotron.
- [ ] If you modified anything related to checkpoints, did you verify that saving and reloading checkpoints still works correctly?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

File: src/nanotron/config/models_config.py [hunk 1]
```diff
@@ -149,6 +149,10 @@ class Qwen2Config:
     no_rope_layer: Optional[
         int
     ] = None  # Skip rope every no_rope_layer layers (see https://arxiv.org/abs/2501.18795 https://arxiv.org/abs/2305.19466 and Llama4)
+    _fused_rotary_emb: bool = False
+    _fused_rms_norm: bool = False
+    _use_qkv_packed: bool = False
```
Comment (NouamaneTazi) [inline src/nanotron/config/models_config.py:154]: these are duplicated

File: src/nanotron/eval/one_job_runner.py [hunk 2]
```diff
@@ -60,6 +60,15 @@ def eval_single_checkpoint(self, uploaded_files: List[dict]) -> Tuple[str, str]:
...
             f"Lighteval Runner got {len(uploaded_files)} files. Using {checkpoint_path} as checkpoint path."
         )
+
+
+        slurm_job_id, slurm_log = run_slurm_one_job(
+            config=self.config,
+            lighteval_config=self.lighteval_config,
+            model_checkpoint_path=checkpoint_path,
+            current_step=self.config.general.step,
+        )
```
Comment (NouamaneTazi) [inline src/nanotron/eval/one_job_runner.py:70]: same

**Review by NouamaneTazi — CHANGES_REQUESTED** [review]
